File: 0643-maximum-average-subarray-i/0643-maximum-average-subarray-i.py
class Solution:
    def findMaxAverage(self, nums: List[int], k: int) -> float:
        left = 0
        window_sum = sum(nums[:k])
        max_avg = window_sum / k
        for i in range(k, len(nums)):
            print(nums[i])
            window_sum -= nums[left]
            window_sum += nums[i]
            max_avg = max(max_avg, window_sum / k)
            left+=1
        return max_avg

File: 0643-maximum-average-subarray-i/0643-maximum-average-subarray-i.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var findMaxAverage = function(nums, k) {
    let sumWindow = nums.slice(0, k).reduce((a, b) => a + b, 0);
    let maxWindow = sumWindow / k;
    let left = 0;
    if (nums.length == k){
        return maxWindow;
    }
    for (let i = k; i < nums.length; i++){
        sumWindow += nums[i] - nums[left];
        left ++;
        let avgWindow = sumWindow / k;
        maxWindow = Math.max(maxWindow, avgWindow);
    }
    return maxWindow
};